app/middleware/auth.py

```diff
@@ -0,0 +1,55 @@
+from fastapi import Depends, Request, HTTPException
+import httpx
+import os
+from enum import StrEnum, auto
+
+
+class UserStatus(StrEnum):
+    isModerator = auto()
+    isLoggedIn = auto()
+
+
+PO_AUTH_ROUTE = os.getenv("PO_AUTH_ROUTE")
+if PO_AUTH_ROUTE is None:
+    raise RuntimeError("PO_AUTH_ROUTE environment variable is not set.")
+
+
+def get_auth_dependency(user_status: UserStatus):
+    async def wrapper(request: Request):
+        return await auth_dependency(request, user_status)
+    return Depends(wrapper)
+
+
+async def auth_dependency(request: Request, user_status: UserStatus):
+    session_cookie = request.cookies.get("session")
+    print("Hello from auth.py")
+
+    if not session_cookie:
+        raise HTTPException(status_code=401, detail="Missing session token")
+
+    if user_status not in UserStatus:
+        raise HTTPException(status_code=400, detail="Invalid user status")
```

Comment:
Be precise, you will appreciate it when you'll bump in the error ;-)
```suggestion
        raise HTTPException(status_code=400, detail=f"Invalid user status: {user_status}")
```